frontstage/views/surveys/add_survey_submit.py

```diff
@@ -49,4 +59,25 @@ def add_survey_submit(session):
     logger.info('Successfully retrieved data for confirm add organisation/survey page',
                 case_id=case_id, party_id=party_id)
     return redirect(url_for('surveys_bp.get_survey_list', _anchor=(business_party_id, added_survey_id), _external=True, business_party_id=business_party_id,
-                            survey_id=added_survey_id, tag='todo'))
+                            survey_id=added_survey_id, tag='todo', already_enrolled=already_enrolled))
+
+
+def is_respondent_and_business_enrolled(associations, survey_id, party_id):
+    """
+    This function checks that the business AND the respondent are enrolled in a survey. We first check if the respondent
+    is enrolled, then if not, we check that the business is enrolled on the survey. A business can have multiple
+    respondents enrolled for a survey, and a respondent can be enrolled for multiple surveys. If we only checked if the
+    business was enrolled, then there could only ever be one respondent that could be able to answer the survey, which
+    isn't the desired functionality.
+
+    :param associations: List of respondents and their enrolled surveys
+    :param survey_id: id of the added survey
+    :param party_id: id of the respondent
+    :return: True if respondent and the business is enrolled on the survey and false otherwise
+    """
+    for association in associations:
+        if association['partyId'] == party_id:
+            for survey in association['enrolments']:
+                if survey['surveyId'] == survey_id:
```

Comment:
```suggestion
          for association in (for association in associations if association['partyId'] == party_id):
            for survey in (for survey in association['enrolments'] if survey['surveyId'] == survey_id):
```

Reply:
I think maybe trying something like this could be the best way to write it.  
```for association in assocations:
  if is_respondent_associated_with_busienss(association, party_id):
    if is_business_enrolled_in_survey(var1, var2)```

Having 2 separate functions that explain in english what they're doing  will be more useful then seeing nested for loops I think.  Then we can have this function tie everything together, along with any explanations that are needed.
This might look awful when implemented?  I don't know, but I think it'd help reduce the complexity of what we're doing here.

Reply:
your suggestion is better than mine

Reply:
I like this approach, not least because splitting functions leads to easier testing, which in turn leads to more accurate testing.